source/slang/slang-lexer.h

```diff
@@ -134,6 +144,7 @@ namespace Slang
 
         TokenFlags      m_tokenFlags;
         LexerFlags      m_lexerFlags;
+        OptionFlags     m_optionFlags;
```

Comment:
I don't think that we need/want to have two different sets of flags that control the lexer's behavior. The existing `LexerFlags` are already being used by code that sits above the lexer to impose higher-level policy on it, and it seems like we can just fold something like a `kLexerFlag_PreserveComments` into the mix without trouble. Or am I missing something that makes the two kinds of flags here different?

Reply:
I actually did have it like this originally, but when I made the change, but found things broke because other code assumed it could just overwrite the contents. So I split out such that 'immutable' options are separate.